docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+
+Under the covers Dojo uses Webpack with a number of custom plugins to provide intelligent application bundling. However, sometimes decisions made by the build tool or manually defined in `.dojorc` can create duplication of common resources shared by multiple bundles. Some of this is unavoidable. A good general rule of thumb for avoid duplication is to try to ensure that common code is at the outermost edges of your dependency tree. In other words, minimize dependencies as much as possible among shared code. If a significant amount of code may be shared among bundles (e.g. common widgets) consider bundling these assets together.
+
+# Assets and Externals
+
+## Assets
+
+Many assets like CSS and images will be imported by modules and inlined automatically by the build process. However, sometimes it is necessary to serve static assets like the favicon or video files.
+
+Static assets can be added to an `assets/` directory at the project root. At build time, these assets are copied to an `assets/` directory along-side the built application.
+
+The build also parses `src/index.html` for CSS, JavaScript, and image assets, hashes them and includes them in the output directory. A favicon can be added to `src` and referenced by `src/index.html`. The build will then hash the file and copy it to the output directory with a file name of `favicon.[hash].ico`.
+
+## Externals
+
+Non-modular libraries or standalone applications that cannot be bundled normally can be included in a Dojo application by providing an implementation of `require` or `define` when needed, and some configuration in the project's `.dojorc` file.
+
+Configuration for external dependencies can be provided under the `externals` property of the `build-app` config. `externals` is an object with two allowed properties:
+
+-   `outputPath`: An optional property specifying an output path to which files should be copied.
+
+-   `dependencies`: A required array that defines which modules should be loaded via the external loader, and what files should be included in the build. Each entry can be one of two types:
+    _ A string that indicates that this path, and any children of this path, should be loaded via the external loader.
+    _ An object that provides additional configuration for dependencies that need to be copied into the built application. This object has the following properties:
+
+| Property | Type                                                    | optional | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
```

Comment:
We should probably be consistent with how we represent the `Optional` columns - I'd suggest we use Yes/No values, similar to the other tables in this guide.